src/dsp_tools/utils/connection_live.py

```diff
@@ -330,7 +333,8 @@ def put(
                 headers=headers,
                 # if data is not encoded as bytes, issues can occur with non-ASCII characters,
                 # where the content-length of the request will turn out to be different from the actual length
-                data=jsondata.encode("utf-8") if jsondata else None,
+                data=json.dumps(jsondata, cls=SetEncoder).encode("utf-8") if jsondata else None,
+                # TODO: ensure_ascii=False?
```

Comment:
please do not check in TODO comments. If it's actually something to do, it should go on Notion or Linear

Reply:
Here I wanted to ask you if `ensure_ascii=False` would resolve the problem described on lines 334-35. You had used it in one single place (https://github.com/dasch-swiss/dsp-tools/blob/0d5627fb5de9f88ef5bab6ee05155a76fc76c791/src/dsp_tools/commands/xmlupload/resource_create_client.py#L44). Or was the problem even caused by `ensure_ascii=False`?

Reply:
Don't the linter prevent todo anymore? Previously I am not sure if it was darglint or pylint that made the checks fail. This could be remedied with this right? https://docs.astral.sh/ruff/rules/#flake8-fixme-fix

Because leaving todo is something that might happen for me accidentally.

Reply:
We could add "FIX" here: https://github.com/dasch-swiss/dsp-tools/blob/d3e238beb2a115f025ce99b7a0cc4861b26feaaf/pyproject.toml#L179-L180

Reply:
works for me

Reply:
> Here I wanted to ask you if `ensure_ascii=False` would resolve the problem described on lines 334-35. You had used it in one single place (
> 
> https://github.com/dasch-swiss/dsp-tools/blob/0d5627fb5de9f88ef5bab6ee05155a76fc76c791/src/dsp_tools/commands/xmlupload/resource_create_client.py#L44
> ). Or was the problem even caused by `ensure_ascii=False`?

`ensure_ascii=True` escapes non-ascii characters with `\xy`, which then might end up in the data, which we don't want. So thinking about it now, we should definitely have it in there. (Or at least experiment with what the exact behaviour is, so we know for sure it does what we want. But I'm pretty sure we want our data to have all UTF-8 characters without escaping.)

the Issue I faced with the encoding seemed more like a byte-array vs. unicode (`b'...'` vs. `'...'`) problem, which messed up the content-length header that's auto-added by requests. I never cared to fully investigate it because that fixed it... but I'm pretty positive that 1) this is something else than the `ensure_ascii` and 2) that it could be solved with using the `json=dict` rather than `data=str` param on the requests (cf. my comment mentioning the notion page).

Reply:
Okay, so if I get you right: I will add `ensure_ascii=False`, and leave the byte encoding in place. Thanks for the clarification :-)

Reply:
> Okay, so if I get you right: I will add `ensure_ascii=False`, and leave the byte encoding in place. Thanks for the clarification :-)

Yes :)